src/home/BudgetAtAGlanceWidget.js

```diff
@@ -1,7 +1,209 @@
 import React from 'react';
+import { useSelector } from 'react-redux';
+import { ProgressBar } from 'react-bootstrap';
+import { PropTypes } from 'prop-types';
+import { INCOME_ID } from '../data';
+import { formatMoney } from '../utilities/index';
+
+/**
+ * Shows how much spent compared to budget across all categories (except income) for current month
+ * Informs user if amount spent is "Under", "On Track", or "Over" budget
+ * Spent percentage is compared to month percentage, status separated by 10%
+ * Displays progress bar
+ *
+ * @component
+ */
+
+function BudgetAtAGlance() {
+  const date = new Date();
+
+  const firstDay = new Date(date.getFullYear(), date.getMonth(), 1);
+
+  const lastDay = new Date(date.getFullYear(), date.getMonth() + 1, 0);
+
+  // Get all transactions that do not fall under income category and reduce
+  const amountSpent = useSelector(
+    (state) => state.transactions.entities
+      .filter((transaction) => transaction.date >= firstDay && transaction.date <= lastDay)
+      .filter((transaction) => transaction.categoryId !== INCOME_ID)
+      .reduce((total, transaction) => total + transaction.amount, 0),
+  );
+
+  // Gets categories data, excludes income category, sums target values
+  const totalBudget = useSelector(
+    (state) => state.categories.entities
+      .filter((category) => category.id !== INCOME_ID)
+      .reduce((total, category) => total + category.target, 0),
+  );
+
+  // Gets budget percentage to calculate status
+  let spentToBudget = '';
+  if (amountSpent >= 0) {
+    spentToBudget = 0;
+  } else {
+    spentToBudget = parseFloat(Math.abs(amountSpent) / totalBudget);
+  }
+
+  // Gets month percentage to calculate status
+  const currDay = date.getDate();
+  const endDay = lastDay.getDate();
+  const monthPercentage = (currDay / endDay);
+
+  let status = '';
+
+  if (spentToBudget >= (monthPercentage + 0.10)) {
+    status = 'Over';
+  } else if (spentToBudget <= (monthPercentage - 0.10)) {
+    status = 'Under';
+  } else {
+    status = 'On Track';
+  }
+
+  return (
+    <div className="row">
+      <div className="col-md-6">
```

Comment:
You know what, I just thought of this. Let's take the row and column divs out of this component and let the HomePage component. handle it. That way the HomePage component can set up rows of various widgets and scale them responsively however it wants to.

Reply:
You don't need to add anything to the HomePage component, we'll set that up later.